src/Adapter/Product/Stock/CommandHandler/UpdateStockHandler.php

```diff
@@ -0,0 +1,99 @@
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\Stock\CommandHandler;
+
+use PrestaShop\PrestaShop\Adapter\Product\Stock\Repository\MovementReasonRepository;
+use PrestaShop\PrestaShop\Adapter\Product\Stock\Update\ProductStockProperties;
+use PrestaShop\PrestaShop\Adapter\Product\Stock\Update\ProductStockUpdater;
+use PrestaShop\PrestaShop\Core\Domain\Product\Stock\Command\UpdateStockCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Stock\CommandHandler\UpdateStockHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Stock\ValueObject\StockModification;
+
+/**
+ * Updates product stock using legacy object model
+ */
+class UpdateStockHandler implements UpdateStockHandlerInterface
+{
+    /**
+     * @var ProductStockUpdater
+     */
+    private $productStockUpdater;
+
+    /**
+     * @var MovementReasonRepository
+     */
+    private $movementReasonRepository;
+
+    /**
+     * @param ProductStockUpdater $productStockUpdater
+     * @param MovementReasonRepository $movementReasonRepository
+     */
+    public function __construct(
+        ProductStockUpdater $productStockUpdater,
+        MovementReasonRepository $movementReasonRepository
+    ) {
+        $this->productStockUpdater = $productStockUpdater;
+        $this->movementReasonRepository = $movementReasonRepository;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function handle(UpdateStockCommand $command): void
+    {
+        $stockModification = null;
+        if ($command->getDeltaQuantity()) {
+            $stockModification = new StockModification(
+                $command->getDeltaQuantity(),
+                $this->movementReasonRepository->getEmployeeEditionReasonId($command->getDeltaQuantity() > 0)
+            );
+        }
+
+        // Now we only fill the properties existing in StockAvailable object model.
+        // Other properties related to stock (which exists in Product object model) should be taken care by a unified UpdateProductCommand.
+        // For now this will also fill some of deprecated properties in product (quantity, location, out_of_stock),
+        // but in future we will remove those fields from Product,
+        // and then this handler will only persist StockAvailable related fields as it is designed for.
+        $this->productStockUpdater->update(
+            $command->getProductId(),
+            new ProductStockProperties(
+                null,
+                $stockModification,
+                $command->getOutOfStockType(),
+                null,
+                $command->getLocation(),
+                null,
+                null,
+                null,
+                null,
+                null
+            ),
+            $command->getShopConstraint()
+        );
```

Comment:
Just an optinal suggestion in case there are other things to fix. This could be moved into a private method `updateProductDeprecatedFields` This way you can understand the purpose quickly without reading the whole comment (even if the comment IS important and it clearly explain in details the purpose here)

Reply:
its not what it does. Deprecated properties are filled inside the StockUpdater, which is still used by both UpdateProductCommand and old UpdateStockInformationCommand.